### ISSUE TITLE
[Open RFP] ( $800 ) - Basic USD/Z export ( Mesh, materials w/ PBR support )

### ISSUE THREAD

**Issue by br-matt** [BitReelCo/BJS-PR-Bounty-Pool#2]
### requirements

- Mesh exported as USD can be loaded into Unreal Editor with Meshes and Materials showing up correctly. PBR Material exported correctly as well. 

- Mesh exported as USDZ can be viewed in AR using iOS built in viewer. Meshes and Materials showing up correctly with PBR or not.

- Documentation in the BJS Docs somewhere

### recommended approach

None

### Notes

Skeletons, Lights, Node Materials and many other things USD supports are out of scope for this first version

**Comment by lokiiarora** [comment]
I would like to take this up :) !
Would a good start be from porting it from Three.js [here](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/exporters/USDZExporter.js)

**Comment by br-matt** [comment]
Hey thanks! That seems like a reasonable start to me. Maybe some of the BJS folks will chime in otherwise though

**Comment by sebavan** [comment]
Seems reasonable to me as well !!! let's see if @bghgary approves the idea ???

**Comment by bghgary** [comment]
I don't see why this won't work.

**Comment by br-matt** [comment]
Hey @lokiiarora just checking in and wondering how this was going ?

**Comment by lokiiarora** [comment]
Hey @br-matt , working on it, I've got to the point where I can export a USDZ file with all textures, geometries and zip working fine.
Haven't specifically gotten the scene file `.usda` working as such

**Comment by br-matt** [comment]
Hey @lokiiarora guessing some things came up. Is it alright if I move this back to Ready status or are you still able to finish it? Not a big deal but want to open it up for someone else in case you became busy

**Comment by Pryme8** [comment]
https://github.com/Pryme8/USDZExport/blob/main/src/exporter/exporter.ts

So Ive got this to the point where it will load nearly the same as the Three.js export.

There are a few TODOs but those are probably out of scope for the initial commit.

One thing that I could use some help with is why some geometry is not loading on the viewer even though the underlying buffers are the same on both exports from Three.js and our export.  Not sure how that works but yeah....

Also don't want to spend a ton of time on this as the bounty would then become not cost effective for my time.

![image](https://github.com/BitReelCo/BJS-PR-Bounty-Pool/assets/1538857/05acad95-5a99-4cb9-a65a-84ffcd7ea2d4)
Ours is on the left, Three.js is on the right.

Pretty sure the color difference is the normal texture being in sRGB space instead of raw.  That is on the list of TODOs that needs to be fixed.

**Comment by br-matt** [comment]
Wow awesome @Pryme8 :) thanks for picking this up instead. @sebavan what do you think maybe forums is best place or here for assistance ?

**Comment by sebavan** [comment]
Yup the PR is up so we can discuss on the PR on the forum. @bghgary  and @popov72 will do a first PR pass there
